rules/scss.js

```diff
@@ -60,4 +63,8 @@ module.exports = {
   'scss/partial-no-import': true,
   // Disallow redundant nesting selectors (`&`).
   'scss/selector-no-redundant-nesting-selector': true,
+  // Disallow dollar variables within a stylesheet.
+  'scss/no-dollar-variables': null,
+  // Disallow duplicate dollar variables within a stylesheet.
+  'scss/no-duplicate-dollar-variables': true,
```

Comment:
This seems like a good enforcement to have. But there are a few places that get flagged in /web.  I don't write enough scss to make this call :p

eg.
```
@mixin logo-width($theme: 'default') {
  $logo-width: 104px;

  @if $theme == 'foo' {
    $logo-width: 140px;
  }

  @if $theme == 'bar' {
    $logo-width: 128px;
  }

  @if $theme == 'baz' {
    $logo-width: 132px;
  }
  width: rem($logo-width);
}
```

Reply:
Nice, you're right, that's a good rule to enforce in most cases, maybe we can relax it with

[`{ "ignoreInsideAtRules": ["if", "mixin"] }`](https://github.com/kristerkari/stylelint-scss/tree/master/src/rules/no-duplicate-dollar-variables#ignoreinsideatrules-array-of-at-rules)